File: morning_exercises/day1.js
function generateSieve(n) {
    let isPrime = new Array(n+1).fill(true);
    isPrime[0] = isPrime[1] = false;

    for(let p=2; p*p<=n; p++) {
        if(isPrime[p]) {
            for (let multiple=p*p; multiple<=n; multiple+=p) {
                isPrime[multiple]=false;
            }
        }
    }
    return isPrime[n];
}

function primeBuzz(n) {
    let isPrime = generateSieve(n);
    let isBuzz = n%5 === 0;

    if (isPrime && isBuzz) console.log("Primebuzz");
    else if (isBuzz) console.log("Buzz");
    else if (isPrime) console.log("Prime");
    else console.log(n);
}

primeBuzz(23)

File: morning_exercises/day1.cs
public static class PrimeSieve
{
    public static bool GenerateSieve(int n)
    {
        bool[] is_prime = new bool[n + 1];
        Array.Fill(is_prime, true);
        is_prime[0] = is_prime[1] = false;

        for (int p = 2; p * p <= n; p++)
        {
            if (is_prime[p])
            {
                for (int multiple = p * p; multiple <= n; multiple += p)
                {
                    is_prime[multiple] = false;
                }
            }
        }

        return is_prime[n];  // return whether n itself is prime
    }
}

public static class Primebuzz
{
    public static void checkPrimebuzz(int n)
    {
        bool isPrime = PrimeSieve.GenerateSieve(n);
        bool isBuzz = n % 5 == 0;

        if (isPrime && isBuzz)
            Console.WriteLine("Primebuzz");
        else if (isBuzz)
            Console.WriteLine("Buzz");
        else if (isPrime)
            Console.WriteLine("Prime");
        else
            Console.WriteLine(n);
    }
}
